src/test/java/reposense/util/TestUtil.java

```diff
@@ -154,16 +155,20 @@ public static boolean compareNumberFilesChanged(int expectedNumberFilesChanged,
         if (gitLogResult.isEmpty()) {
             return expectedNumberFilesChanged == 0;
         }
-        String[] changesLogged = gitLogResult.split("\n");
+        String[] changesLogged = gitLogResult.split(GitLog.COMMIT_INFO_DELIMITER);
         HashSet<String> filesChanged = new HashSet<>();
-        // Checks the 2nd line of each commit which contains info of the changed file
-        for (int i = 0; i < changesLogged.length; i += 4) {
-            String log = changesLogged[i + 1];
-            String fileChanged = log.split("\\| ")[0].trim();
-            if (fileChanged.contains("=>")) {
-                fileChanged = fileChanged.substring(fileChanged.indexOf("=> ") + 3);
+
+        // start from index 1 as index 0 is always empty.
+        for (int i = 1; i < changesLogged.length; i++) {
+            String[] commitInfo = changesLogged[i].replaceAll("\n+$", "").split("\n");
+            int fileChangedNum = Integer.parseInt(commitInfo[commitInfo.length - 1].trim().split(" ")[0]);
+            for (int fileNum = 0; fileNum < fileChangedNum; fileNum++) {
+                String fileChanged = commitInfo[commitInfo.length - 2 - fileNum].split("\\| ")[0].trim();
+                if (fileChanged.contains("=>")) {
+                    fileChanged = fileChanged.substring(fileChanged.indexOf("=> ") + 3);
+                }
+                filesChanged.add(fileChanged);
```

Comment:
The complexity of this method seems to be getting unreadable.
Is it possible to shorten it using StringsUtil#filterText to look and count the number of headers?

Reply:
Don't think it's possible as `StrungsUtil#filterText` performs different functions, different from our requirements.

Reply:
Yes, break it into smaller methods, e.g. getFilesChangedInCommit(...) so this is for just a single chunk of "commit message"